### PR TITLE
Modify `databricks_repo` to support Git folders in the workspace

### PR DESCRIPTION
**Pull request by alexott**
## Changes

Changes include:

* adjust `path` check to enforce it only on `/Repos`, not everywhere.
* adjust documentation

## Tests

- [x] `make test` run locally
- [x] relevant change in `docs/` folder
- [x] covered with integration tests in `internal/acceptance`
- [x] relevant acceptance tests are passing
- [ ] using Go SDK

### CODE REVIEW

File: docs/resources/repo.md [hunk 1]
```diff
@@ -3,13 +3,13 @@ subcategory: "Workspace"
...
+-> **Note** To create a Git folder from a private repository you need to configure Git token as described in the [documentation](https://docs.databricks.com/en/repos/index.html#configure-your-git-integration-with-databricks).  To set this token you can use [databricks_git_credential](git_credential.md) resource.
 
 ## Example Usage
 
-You can declare Terraform-managed Repo by specifying `url` attribute of Git repository. In addition to that you may need to specify `git_provider` attribute if Git provider doesn't belong to cloud Git providers (Github, GitLab, ...).  If `path` attribute isn't provided, then repo will be created in the user's repo directory (`/Repos/<username>/...`):
+You can declare Terraform-managed Git folder by specifying `url` attribute of Git repository. In addition to that you may need to specify `git_provider` attribute if Git provider doesn't belong to cloud Git providers (Github, GitLab, ...).  If `path` attribute isn't provided, then Git folder will be created in the user's directory (for existing workspaces it's `/Repos/<username>/...`, for new workspaces it will be user's folder in the workspace):
```
Comment (dbbnicole) [inline docs/resources/repo.md:12]: @damiangajd-db Has the Repos Create API default location been changed? If not, we should time the merge of this Pr with that change.

File: docs/resources/repo.md [hunk 2]
```diff
@@ -20,30 +20,30 @@ resource "databricks_repo" "nutter_in_home" {
...
 
-* `url` -  (Required) The URL of the Git Repository to clone from. If the value changes, repo is re-created.
+* `url` -  (Required) The URL of the Git Repository to clone from. If the value changes, Git folder is re-created.
 * `git_provider` - (Optional, if it's possible to detect Git provider by host name) case insensitive name of the Git provider.  Following values are supported right now (could be a subject for a change, consult [Repos API documentation](https://docs.databricks.com/dev-tools/api/latest/repos.html)): `gitHub`, `gitHubEnterprise`, `bitbucketCloud`, `bitbucketServer`, `azureDevOpsServices`, `gitLab`, `gitLabEnterpriseEdition`, `awsCodeCommit`.
-* `path` - (Optional) path to put the checked out Repo. If not specified, then repo will be created in the user's repo directory (`/Repos/<username>/...`).  If the value changes, repo is re-created.
+* `path` - (Optional) path to put the checked out Git folder. If not specified, , then the Git folder will be created in the user's directory (for existing workspaces it's `/Repos/<username>/...`, for new workspaces it will be user's folder in the workspace).  If the value changes, Git folder is re-created.
```
Comment (dbbnicole) [inline docs/resources/repo.md:29]: Also watch for timing of release

**Comment by alexott** [comment]
@dbbnicole if it's only text, we can split this into 2 PRs to unblock usage of git folders...

File: docs/resources/repo.md (hunk 1 above)
Comment (damiangajd-db) [reply docs/resources/repo.md:12]: We haven't changed it yet.

Comment (alexott) [reply docs/resources/repo.md]: @dbbnicole @damiangajd-db I've updated description - please approve if it's ok

**Pull request merged**